clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -126,6 +144,11 @@ public OutputStream wrapForOutput(ByteBufferOutputStream buffer, byte messageVer
         public InputStream wrapForInput(ByteBuffer buffer, byte messageVersion, BufferSupplier decompressionBufferSupplier) {
             return ZstdFactory.wrapForInput(buffer, messageVersion, decompressionBufferSupplier);
         }
+
+        @Override
+        public int getRecommendedDOutSize() {
+            return 16 * 1024; // 16KB
```

Comment:
We decided not to get this info from the zstd library?

Reply:
This will be picked up as a separate follow-up PR: https://issues.apache.org/jira/browse/KAFKA-14634 

I decided to keep it separate is to introduce fewer changes at a time and measure their impact.

Reply:
The PR says:

> we pushed the skipping of key/value logic to zstd-jni implementation instead of using the one provided by BufferedInputStream

Reply:
OK, that's fine.

Reply:
Also, can you please remind me if the 16 KB number is different from what was there before for this case?

Reply:
It was 16KB itself (+2KB for skipBuffer which we removed). https://github.com/apache/kafka/blob/ef09a2e3fc11a738f6681fd57fb84ad109593fd3/clients/src/main/java/org/apache/kafka/common/compress/ZstdFactory.java#L68